src/cpp/fastdds/builtin/type_lookup_service/TypeLookupReplyListener.cpp

```diff
@@ -53,11 +52,68 @@ TypeLookupReplyListener::~TypeLookupReplyListener()
 {
 }
 
+void TypeLookupReplyListener::check_get_types_reply(
+        const SampleIdentity& request_id,
+        const TypeLookup_getTypes_Out& reply)
+{
+    // Check if we were waiting that reply SampleIdentity
+    auto requests_it = typelookup_manager_->async_get_type_requests_.find(request_id);
+    if (requests_it != typelookup_manager_->async_get_type_requests_.end())
+    {
+        for (xtypes::TypeIdentifierTypeObjectPair pair : reply.types())
+        {
+            if (RETCODE_OK == fastrtps::rtps::RTPSDomainImpl::get_instance()->type_object_registry_observer().
+                            register_type_object(pair.type_identifier(), pair.type_object()))
+            {
+                typelookup_manager_->notify_callbacks(requests_it->second);
+            }
+        }
+        // Erase the processed SampleIdentity
+        typelookup_manager_->remove_async_get_types_request(request_id);
+    }
+}
+
+void TypeLookupReplyListener::check_get_type_dependencies_reply(
+        const SampleIdentity& request_id,
+        const fastrtps::rtps::GuidPrefix_t type_server,
+        const TypeLookup_getTypeDependencies_Out& reply)
+{
+    // Check if we were waiting that reply SampleIdentity
+    auto requests_it = typelookup_manager_->async_get_type_requests_.find(request_id);
+    if (requests_it != typelookup_manager_->async_get_type_requests_.end())
+    {
+        bool all_dependencies_known = true;
+        for (xtypes::TypeIdentfierWithSize type_id : reply.dependent_typeids())
+        {
+            if (RETCODE_OK != typelookup_manager_->check_type_identifier_received(type_id, type_server, nullptr))
+            {
+                all_dependencies_known = false;
+            }
+        }
+
+        // If all dependencies are known, ask for the parent type
+        if (all_dependencies_known)
+        {
+            xtypes::TypeIdentifierSeq uknown_type;
+            uknown_type.push_back(requests_it->second.type_id());
+
+            SampleIdentity get_types_request = typelookup_manager_->get_types(uknown_type, type_server);
+            if (builtin::INVALID_SAMPLE_IDENTITY != get_types_request)
+            {
+                // Store the sent request and associated TypeIdentfierWithSize
+                typelookup_manager_->add_async_get_type_request(get_types_request, requests_it->second);
+            }
+        }
+        // Erase the processed SampleIdentity
+        typelookup_manager_->remove_async_get_types_request(request_id);
+    }
+}
```

Comment:
I do not think the implementation of this method is correct.

* If there is some dependency unknown, that dependency should also be included into the  the get_types request.
* This method is not checking the continuation point. In case that the continuation point is set, another get_dependencies request with this new continuation point must be sent.